#1746: Added MissionStart logging to audit and validation missions

wilsmic2 opened the pull request:
Resolves #1746 

Added logging for when users start an auditing mission with the name "MissionStart" along with the missionId, missionType, distanceMeters, and regionId.

Changed the logging syntax for when users start a validation mission to be aligned with the logging message when they finish a validation mission.

wilsmic2 commented on public/javascripts/SVLabel/src/SVLabel/modal/ModalMission.js:
```diff
@@ -63,6 +63,18 @@ function ModalMission (missionContainer, neighborhoodContainer, uiModalMission,
 
     this._handleCloseButtonClick = function () {
+        mission = _missionContainer.getCurrentMission();
+        if(mission.getProperty("distanceProgress") < 0.0001) {
```
I added this check because if a user was halfway through a mission and then came back to resume it later, it would log MissionStart even though they are already into the mission.

misaugstad replied:
Ahh makes sense!

jonfroehlich commented:
Maybe add that comment to the code itself since it wasn’t immediately obvious. 

Sent from my iPhone

> On Jul 8, 2019, at 12:21 PM, Mikey Saugstad <notifications@github.com> wrote:
> 
> @misaugstad commented on this pull request.
> 
> In public/javascripts/SVLabel/src/SVLabel/modal/ModalMission.js:
> 
> > @@ -63,6 +63,18 @@ function ModalMission (missionContainer, neighborhoodContainer, uiModalMission,
>      };
>  
>      this._handleCloseButtonClick = function () {
> +        mission = _missionContainer.getCurrentMission();
> +        if(mission.getProperty("distanceProgress") < 0.0001) {
> Ahh makes sense!
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

wilsmic2 commented:
Done! @jonfroehlich

jonfroehlich commented:
👍🏽

Sent from my iPhone

> On Jul 8, 2019, at 2:44 PM, wilsmic2 <notifications@github.com> wrote:
> 
> Done! @jonfroehlich
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

misaugstad commented:
1. Could you have the validation page have the same names for mission start/complete? So do "MissionStart" for each, not "ClickOk_MissionStart", the clickOK isn't important
1. I am not seeing a mission complete being logged on the validation page. Is that being logged for you?
1. When logging mission start/complete on validation, could you add in the mission id, mission type, label type, and labels validated? This will better mirror the audit page.

wilsmic2 commented:
1. Yes! I wasn't sure if I should keep the original formatting for that one (being the ClickOk_MissionStart) but I have changed it now to be consistent with the auditing logs.
2. I originally had it being logged after users clicked the "validate more labels" button after they finish the mission, but I have changed it to be more consistent with the auditing MissionComplete by having it be logged when the mission summary screen is displayed.
3. Yes!

misaugstad commented on public/javascripts/SVValidate/src/modal/ModalMission.js:
```diff
@@ -16,7 +16,21 @@ function ModalMission (uiModalMission, user) {
-        svv.tracker.push("ModalMission_ClickOK");
+        var mission = svv.missionContainer.getCurrentMission();
+
+        // Check added so that if a user begins a mission, leaves partwar through, and then resumes the mission later,
```
partwar -> partway

wilsmic2 commented:
Ah dang I see that now! I didn't notice it at first since they were spread out in the database for some reason. This should fix it since I believe that the onClick event was being assigned to the function twice, so this should make sure that it is only done once!
